action.yaml

```
@@ -24,7 +24,8 @@ runs:
         echo ::group::Install dependencies
         # Remove this package from constraints
         PKG_NAME=$(pip list --local --editable --format json | jq '.[0].name' -r)
-        grep -v "^${PKG_NAME}@" ${GITHUB_ACTION_PATH}/constraints.txt > constraints.txt
+        cp -a ${GITHUB_ACTION_PATH}/constraints{,-heads,-pins}.txt ./
+        grep -iv "^${PKG_NAME}@" constraints-heads.txt
```

I don't think that this actually writes the result anymore. The `-i` flag on grep governs case-insensitivity rather than inline replacement as it does in sed.

```suggestion
        sed -i "/^${PKG_NAME}@/d" constraints-heads.txt
```

---

Ya know, this is why we review PRs like this.

I actually made the same mistake on my dev branch and then fixed it but forgot to bring the fix here.

A slight complication is macOS, where sed seems to Think Different with regards to `-i`. This syntax seems to work:
```suggestion
        sed -i'' -e "s/^${PKG_NAME}@.*//g" constraints-heads.txt
```

---

Not surprised that non-GNU (MacOS userland is from FreeBSD) requires an argument for -i as the flag is meant to indicate the extension of the backup file. It's worth confirming that this does behave as expected (if the backup file extension is empty does that get handled as no-backup (the case when no suffix is supplied for GNU sed) or does it "backup" the unedited file to its original path thus causing no edit to be made. An alternative solution would be to use `-i.bak` and just let the backup file litter the workspace.

---

I went ahead and added `-i.orig` to the sed. A backup of that file will do no harm here.